dusty/daemon.py

```diff
@@ -42,6 +42,14 @@ def _refresh_warnings():
         refresh_config_warnings()
         refresh_preflight_warnings()
 
+def _run_pre_command_functions(connection, suppress_warnings, client_version):
+    _refresh_warnings()
+    _send_warnings_to_client(connection, suppress_warnings)
+    if client_version != VERSION:
+        raise RuntimeError("Dusty daemon is running version: {}, and client is running version: {}".format(VERSION, client_version))
```

Comment:
Version check should be done before any of the other stuff, and it should probably be separate from the rest of the functionality we'll stuff in this function